library/std/src/sys/net/connection/uefi/tcp4.rs

```diff
@@ -67,6 +68,29 @@ impl Tcp4 {
         if r.is_error() { Err(crate::io::Error::from_raw_os_error(r.as_usize())) } else { Ok(()) }
     }
 
+    pub(crate) fn get_mode_data(&self) -> io::Result<tcp4::ConfigData> {
+        // Using MaybeUninit::uninit() generates a Page Fault Here
+        let mut config_data: MaybeUninit<tcp4::ConfigData> = MaybeUninit::zeroed();
```

Comment:
nit: Could you use [`ConfigData::default()`](https://docs.rs/r-efi/latest/r_efi/protocols/tcp4/struct.ConfigData.html#impl-Default-for-ConfigData) here instead of `MaybeUninit`? Slightly simpler, and then the `unsafe` `assume_init` isn't needed.

Reply:
Yes, but it should only be used if necessary for performance. I wouldn't think that's a concern here, so I'd rather have simpler code and one fewer unsafe blocks.

Reply:
> Well, isn't `MaybeUninit` designed to be used in such cases?

Technically yes it works here, but if there's a better option that reduces `unsafe` then there isn't any reason not to use it.

With `MaybeUninit::zeroed()` and later calling `assume_init()`, the assumption is made that zero is a valid bitpattern for all fields that `(*protocol).get_mode_data` doesn't overwrite. This is a nontrivial interaction and really be in a `SAFETY` comment. Also apparently `get_mode_data` expects at least partially initialized data if there is a page fault with `uninit` (presumably trying to read the pointer if nonnull?).

If you use `default()` then all these assumptions can go away. It's also more robust against adding/changing fields across a crate version bump.

Reply:
Ok, switched to using default. Also tested on ovmf.